File: leetcode/easy/9.Palindrome Number/python/palidromeNumber.py
"""Palindrome Number"""

# My Approach: Cast to String and Two Technique Pointer
def isPalindrome(self, x: int) -> bool:
        x = str(x)
        left_pointer = 0
        right_pointer = len(x) - 1
        while left_pointer < right_pointer:
            if x[left_pointer] != x[right_pointer]:
                return False
            left_pointer += 1
            right_pointer -= 1
        return True

'''Complexity Analysis:
Time Complexity: O(N)
Space Complexity: O(1)'''

File: leetcode/easy/9.Palindrome Number/js/palidromeNumber.js
"use strict";

// Palindrome Number
/**
 * @param {number} x
 * @return {boolean}
 */
//My Approach: Cast to String and then Two Pointers Techique
var isPalindrome = function(x) {
    x = String(x);
    let leftPointer = 0;
    let rightPointer = x.length - 1;
    while (leftPointer < rightPointer) {
        if (x[leftPointer] !== x[rightPointer]) {
            return false;
        };
        leftPointer++;
        rightPointer--;
    };
    return true;
};

// Complexity Analysis:
// Time Complexity: O(N)
// Space Complexity: O(1)